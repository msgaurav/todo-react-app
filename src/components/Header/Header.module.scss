.Header_wrapper {
    text-align: center;
    padding: 1rem;
    background-color: steelblue;
    color: white;
    font-size: 1.5rem;

    h1 {
        margin: 1rem;
    }

    @media only screen and (max-width: 768px) {
        font-size: 1rem;
    }
}