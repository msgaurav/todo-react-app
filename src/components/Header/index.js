import React from 'react'
import PropTypes from 'prop-types'
import styles from './Header.module.scss'

export default function Header({ headerData }) {
  const { heading, description } = headerData

  return (
    <div className={styles.Header_wrapper}>
      <h1>{heading}</h1>
      <p>{description}</p>
    </div>
  )
}

Header.defaultProps = {
  headerData: PropTypes.shape({
    description: '',
  }),
}

Header.propTypes = {
  headerData: PropTypes.shape({
    heading    : PropTypes.string,
    description: PropTypes.string,
  }),
}
