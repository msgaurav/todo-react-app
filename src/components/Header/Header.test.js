import React from 'react'
import { render } from '@testing-library/react'
import Header from './index'

const propsData = {
  heading    : 'ToDo App',
  description: 'Quick to add tasks, organize them into lists',
}

describe('The Header Component', () => {
  it('snapshot renders correctly', () => {
    const { asFragment } = render(
      <Header
        headerData={propsData}
      />,
    )
    expect(asFragment()).toMatchSnapshot()
  })

  it('should renders with correct heading', () => {
    const { getByText } = render(
      <Header
        headerData={propsData}
      />,
    )
    expect(getByText('ToDo App')).toBeInTheDocument()
  })

  it('should renders with correct description', () => {
    const { queryByText } = render(
      <Header
        headerData={propsData}
      />,
    )
    expect(
      queryByText('Quick to add tasks, organize them into lists'),
    ).toBeTruthy()
  })
})
