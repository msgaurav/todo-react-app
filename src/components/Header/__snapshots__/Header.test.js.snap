// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Header Component snapshot renders correctly 1`] = `
<DocumentFragment>
  <div
    class="Header_wrapper"
  >
    <h1>
      ToDo App
    </h1>
    <p>
      Quick to add tasks, organize them into lists
    </p>
  </div>
</DocumentFragment>
`;
